### PR TITLE
    Support Pi05 SFT training with DDP

### PR DESCRIPTION
**Pull request by XueSongTap**
## Is your feature request related to a problem? Please describe.
  Currently LoongForge lacks complete out-of-the-box support for Pi0.5 SFT training with the DDP backend. Users who want to
  fine-tune Pi0.5 need to manually prepare checkpoints, adapt launch arguments, and infer the expected training setup from
  scattered model/configuration code.

## Describe the solution you'd like
  This PR adds Pi0.5 SFT training support with the DDP backend and provides the required checkpoint conversion workflow.

  Main changes:
  - Add Pi0.5 HuggingFace safetensors to Megatron torch checkpoint conversion script:
    - `tools/convert_checkpoint/pi05/convert_hf_to_torch.py`
  - Add a user-facing conversion wrapper:
    - `examples/pi05/checkpoint_convert/convert_pi05_hf_to_torch.sh`

  - Update Pi0.5 SFT launch script:
    - `examples/pi05/finetuning/sft_pi05.sh`
    - Switches the training path to DDP distributed training.
    - Updates batch settings for 8-GPU default usage.
    - Keeps precision-aware optimizer configuration for Pi0.5 SFT.
  - Update Pi0.5 configuration:
    - `loongforge/models/embodied/pi05/configuration_pi05.py`
    - Adds optional `param_sync_func` and `grad_sync_func` hooks
  - Add Pi0.5 training documentation:
    - `docs/source/vla_tutorial/quick_start_pi05_training.md`
    - Documents LeRobot v3.0 dataset expectation, checkpoint conversion, and SFT launch parameters.
  - Replace the placeholder VLA quick start page with Pi0.5 quick start content:
    - `docs/source/vla_tutorial/quick_start_vla_training.md`

  Expected usage flow:
  1. Prepare Pi0.5 HuggingFace checkpoint and LeRobot v3.0 dataset.
  2. Convert HF safetensors checkpoint to LoongForge/Megatron torch checkpoint:
```bash
     export LOAD=/path/to/pi05_huggingface/
     export SAVE=/path/to/pi05_torch/
     sh examples/pi05/checkpoint_convert/convert_pi05_hf_to_torch.sh
```
  3. Launch Pi0.5 SFT training with DDP backend:
```
     bash examples/pi05/finetuning/sft_pi05.sh
```

## Describe alternatives you've considered
  I considered using the FSDP training path for Pi0.5 SFT. LoongForge now supports both FSDP and DDP backends for this
  model, but this PR makes DDP the default example path because it is simpler to launch, easier to debug, and better
  aligned with the current Pi0.5 SFT quick-start workflow.

  The FSDP path can still be used when users need sharded data-parallel training, while the DDP path provides a more
  straightforward default for getting Pi0.5 SFT running.

## Test Result：
training with ddp
image
loss precision compare with lerobot
image

### CODE REVIEW

**Comment by nullnonenilNULL** [comment]
Please provide a detailed explanation of the main purpose of the current submission and the test results; you can refer to our PR template: https://github.com/baidu-baige/LoongForge/blob/master/.github/ISSUE_TEMPLATE/feature_request.md
